### PR TITLE
[federated_plugin] add iOS implementation

### PR DESCRIPTION
**Pull request by AyushBherwani1998**
## Description

Adds iOS implementation for federated_plugin

cc:// @RedBrogdon @domesticmouse  I don't have Apple Developer account to test it on my iPhone, and on simulator it throws exception. I read that you can get battery level of simulator on reddit. It would be great if you can check it when ever you get time.

### CODE REVIEW

File: experimental/federated_plugin/federated_plugin/ios/Classes/SwiftFederatedPlugin.swift [hunk 1]
```diff
@@ -3,12 +3,24 @@ import UIKit
...
   public static func register(with registrar: FlutterPluginRegistrar) {
-    let channel = FlutterMethodChannel(name: "federated_plugin", binaryMessenger: registrar.messenger())
+    let channel = FlutterMethodChannel(name: "battery", binaryMessenger: registrar.messenger())
     let instance = SwiftFederatedPlugin()
     registrar.addMethodCallDelegate(instance, channel: channel)
   }
 
   public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
-    result("iOS " + UIDevice.current.systemVersion)
+    guard  call.method == "getBatteryLevel"  else {
```
Comment (RedBrogdon) [inline experimental/federated_plugin/federated_plugin/ios/Classes/SwiftFederatedPlugin.swift:12]: nit: it looks like there are some extra whitespace characters here. Is that intentional?

**Review by RedBrogdon — APPROVED** [review]
LGTM, though I also lack a hardware iOS device to test with.

**Comment by jmagman** [comment]
Tried on a real device, it works (`device.batteryState` returns `full`)
Screen Shot 2020-11-02 at 2 10 54 PM

**Comment by RedBrogdon** [comment]
Much obliged, @jmagman!

File: experimental/federated_plugin/federated_plugin/ios/Classes/SwiftFederatedPlugin.swift (hunk 1 above)
Comment (AyushBherwani1998) [reply experimental/federated_plugin/federated_plugin/ios/Classes/SwiftFederatedPlugin.swift:12]: Opps! I'll fix this in the next commit for MacOS support.

**Pull request merged**